portfolio/src/main/webapp/script.js

```diff
@@ -11,17 +11,38 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+/*
+  Controls slideshow on page
+*/
+var slideIndex = 1;
```

Comment:
In general having a global variable like this is not a good practice. It means we put it in the global scope and might cause conflicts or leaks in the future.

Using a class might be better

```
class slideShower {
  constructor() {
    this.currentIndex = 1;
    this.slides = document.getElementsByClassName("slides");
    this.dots = document.getElementByClassName("dot");
  }

  plusSlides(number) {
    this.currentIndex += number;
    showSlides();
  }
  
  showSlides() {
    ...
  }
}

const shower = new slideShower();
```